lua/lspconfig/vdmj.lua

```diff
@@ -0,0 +1,99 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = 'vdmj'
+
+local function getJarPath(config, package, version)
+  return util.path.join(config.options.mavenrepo, package, version, package .. '-' .. version .. '.jar')
+end
+
+local function get_version(config)
+  if config.options.high_precision == true then
+    return config.options.version:gsub('([%d.]+)', '%1-P')
+  else
+    return config.options.version
+  end
+end
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'java' },
+    filetypes = { 'vdm' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.fn.getcwd()
+    end,
+    options = {
+      java = tostring(vim.fn.getenv 'JAVA_HOME') or '/usr' .. '/bin/java',
+      java_opts = { '-Xmx3000m', '-Xss1m' },
+      mavenrepo = util.path.join(tostring(vim.fn.getenv 'HOME'), '/.m2/repository/com/fujitsu'),
+      logfile = util.path.join(vim.fn.stdpath 'cache', 'vdm-lsp.log'),
+      debugger_port = 0,
+      high_precision = false,
+    },
+  },
+  docs = {
+    description = [[
+https://github.com/nickbattle/vdmj
+
+The VDMJ language server can be installed by cloning the VDMJ repository and
+running `mvn clean install`.
+
+Various options are provided to configure the language server, with the
+following being mandatory:
+
+- `version`,
+- `dialect` (one of `vdmsl`, `vdmpp` or `vdmrt`).
+
+```lua
+require'lspconfig'.vdmj.setup {
+  options = {
+    version = '4.4.3-SNAPSHOT',
+    dialect = 'vdmsl',
+    ...
+  }
+  ...
+}
+```
+
+A value of `0` for `debugger_port` will disable the debugger. Note that if a
+non-zero value is used, only one instance of the server can be active at a
+time.
+
+Note: proof obligations and combinatorial testing are not currently supported
+by neovim.
+]],
+    default_config = {
+      cmd = "Generated from the options given",
+      root_dir = "Git root or vim's starting directory",
+      options = {
+        java = '$JAVA_HOME/bin/java',
+        mavenrepo = '$HOME/.m2/repository/com/fujitsu',
+        logfile = "path.join(vim.fn.stdpath 'cache', 'vdm-lsp.log')",
+        debugger_port = 0,
+        high_precision = false,
+      },
+    },
+  },
+  on_new_config = function(config)
+    local version = get_version(config)
+    config.cmd = {
+      config.options.java,
+      -- TODO: how to unpack java_opts here? unpack(config.options.java_opts)
+      -- results in only the first option being used.
+      '-Xmx3000m',
+      '-Xss1m',
```

Comment:
As per comment, I would like to give the user the ability to override the default Java options.